pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -468,6 +468,13 @@ void downloadAndReplaceSegment(String segmentName, SegmentZKMetadata segmentZKMe
 
   private void downloadSegmentFromDeepStore(String segmentName, IndexLoadingConfig indexLoadingConfig, String uri) {
     File segmentTarFile = new File(_indexDir, segmentName + TarGzCompressionUtils.TAR_GZ_FILE_EXTENSION);
+    if (segmentTarFile.exists()) {
```

Comment:
should we make this part of `SegmentFetcherFactory.fetchSegmentToLocal(uri, segmentTarFile, overwrite=true);` ?

Reply:
hmm, possible. but it's used in a bunch of places, and I don't want to make a call for what each method should do (maybe one of it is relying on the failure)